src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1643,5 +1651,84 @@ private bool InvocableNameofInScope()
             lookupResult.Free();
             return result;
         }
+
+#nullable enable
+        private BoundFunctionPointerInvocation BindFunctionPointerInvocation(SyntaxNode node, BoundExpression boundExpression, AnalyzedArguments analyzedArguments, DiagnosticBag diagnostics)
+        {
+            RoslynDebug.Assert(boundExpression.Type is FunctionPointerTypeSymbol);
+
+            var funcPtr = (FunctionPointerTypeSymbol)boundExpression.Type;
+
+            var overloadResolutionResult = OverloadResolutionResult<FunctionPointerMethodSymbol>.GetInstance();
+            HashSet<DiagnosticInfo>? useSiteDiagnostics = null;
+            var methodsBuilder = ArrayBuilder<FunctionPointerMethodSymbol>.GetInstance(1);
+            methodsBuilder.Add(funcPtr.Signature);
+            OverloadResolution.FunctionPointerOverloadResolution(
+                methodsBuilder,
+                analyzedArguments,
+                overloadResolutionResult,
+                ref useSiteDiagnostics);
+
+            if (!overloadResolutionResult.Succeeded)
+            {
+                overloadResolutionResult.ReportDiagnostics(
+                    binder: this,
+                    node.Location,
+                    nodeOpt: null,
+                    diagnostics,
+                    name: null,
+                    boundExpression,
+                    boundExpression.Syntax,
+                    analyzedArguments,
+                    methodsBuilder.ToImmutableAndFree(),
+                    typeContainingConstructor: null,
+                    delegateTypeBeingInvoked: null,
+                    functionPointerMethodBeingInvoked: funcPtr.Signature,
+                    returnRefKind: funcPtr.Signature.RefKind);
+
+                return new BoundFunctionPointerInvocation(
+                    node,
+                    boundExpression,
+                    funcPtr,
+                    analyzedArguments.Arguments.SelectAsArray((expr, args) => args.binder.BindToNaturalType(expr, args.diagnostics), (binder: this, diagnostics)),
+                    analyzedArguments.RefKinds.ToImmutableOrNull(),
+                    CreateErrorType(),
```

Comment:
>CreateErrorType() [](start = 20, length = 17)

```funcPtr.Signature.ReturnType```? Since there is only one candidate, we can use return type for error recovery.